### PR TITLE
Add unit test for host component

### PR DESCRIPTION
**Pull request by a2batic**
Current code coverage : 24.87%
Tendrl-bug-id: #862
Signed-off-by: a2batic <murarkakanika@gmail.com>

### CODE REVIEW

File: test/unit/host-list-test.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            nodeStore = _nodeStore_;
+            hostList = _hostList_;
+            capitalizeFilter = _capitalizeFilter_;
+        });
+
+    });
+
+    beforeEach(function() {
+
+
```
Comment (gnehapk) [inline test/unit/host-list-test.js]: @a2batic Remove extra lines.

File: test/unit/host-list-test.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        it("Should cancel the timer", function() {
+            sinon.stub($interval, "cancel");
+            $scope.$destroy();
+
+            expect($interval.cancel.calledOnce).to.be.true;
+        });
+
+        it("Should listen GotClusterData event broadcast", function() {
+            //sinon.stub($state, "go");
```
Comment (gnehapk) [inline test/unit/host-list-test.js]: @a2batic Remove commented lines if not required.

File: test/unit/host-list-test.js [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+            $rootScope.clusterData = [];
+            $scope.$broadcast("GotClusterData");
+            expect($state.go.calledWith("clusters")).to.be.true;
+
+            $rootScope.clusterData = ["cluster1"];
+            $scope.$broadcast("GotClusterData");
+            expect($state.go.calledOnce).to.be.false;
+        });
+
+
```
Comment (gnehapk) [inline test/unit/host-list-test.js:161]: @a2batic Remove extra lines.

File: test/unit/host-list-test.js [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+        getHostListDeferred.resolve(hostList.nodes);
+        $rootScope.$digest();
+        vm.hostList.forEach(function(o) { delete o.$$hashKey });
+        expect(vm.filtersText).to.be.equal("Name : 27\n");
+        expect(vm.filteredHostList).to.deep.equal(hostList.filteredNameFormattedOutput);
+    });
+
+    it("Should filter the list with 'status' parameters", function() {
+        vm = $componentController("hostList", { $scope: $scope });
+
```
Comment (gnehapk) [inline test/unit/host-list-test.js]: @a2batic Remove extra lines.

File: test/unit/host-list-test.js [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+        vm.sortConfig.isAscending = false;
+        getHostListDeferred.resolve(hostList.nodes);
+        $rootScope.$digest();
+        vm.hostList.forEach(function(o) { delete o.$$hashKey });
+        expect(vm.hostList).to.deep.equal(hostList.sortedformattedOutput);
+    });
+
+    it("Should filter the list with 'name' parameters", function() {
+        vm = $componentController("hostList", { $scope: $scope });
+
```
Comment (gnehapk) [inline test/unit/host-list-test.js]: @a2batic Remove extra lines.

**Review by gnehapk — APPROVED** [review]

**Review by r0h4n — APPROVED** [review]

**Pull request merged**